core/src/test/scala/com/salesforce/op/stages/impl/regression/RegressionModelSelectorTest.scala

```diff
@@ -213,6 +216,53 @@ class RegressionModelSelectorTest extends FlatSpec with TestSparkContext
     justScores.length shouldEqual transformedData.count()
   }
 
+  it should "fit and predict for even when some models fail" in {
+    val testEstimator = RegressionModelSelector
+      .withCrossValidation(
+        numFolds = 4,
+        validationMetric = Evaluators.Regression.mse(),
+        seed = 10L,
+        modelTypesToUse = Seq(RegressionModelsToTry.OpLinearRegression,
+          RegressionModelsToTry.OpGeneralizedLinearRegression)
+      )
+      .setInput(label, features)
+
+
+    val model = testEstimator.fit(data)
+    model.evaluateModel(data)
+
+    // evaluation metrics from train set should be in metadata
+    val metaData = ModelSelectorSummary.fromMetadata(model.getMetadata().getSummaryMetadata())
+    RegressionEvalMetrics.values.foreach(metric =>
+      assert(metaData.trainEvaluation.toJson(false).contains(s"${metric.entryName}"),
+        s"Metric ${metric.entryName} is not present in metadata: " + metaData)
+    )
+    metaData.validationResults.foreach(println(_))
+    metaData.validationResults.size shouldBe 42
+  }
+
+
+  it should "fail when all models fail" in {
```

Comment:
can you explain in comments or assertions which model fails and why in each case?

Reply:
maxWait could be tested explicitly with some short duration.